src/snowflake/snowpark/dataframe.py

```diff
@@ -707,24 +711,52 @@ def select(
 
             >>> df_selected = df.select(df["col1"], df.col2, df.col("col3"))
 
+        Example 5::
+
+            >>> from snowflake.snowpark.functions import table_function
+            >>> split_to_table = table_function("split_to_table")
+            >>> df_selected = df.select(df.col1, split_to_table(df.col2, lit(" ")), df.col("col3"))
+
         Args:
-            *cols: A :class:`Column`, :class:`str`, or a list of those.
+            *cols: A :class:`Column`, :class:`str`, :class:`TableFunctionCall`, or a list of those. Note that at most one
+                   :class:`TableFunctionCall` object is supported within a select statement.
         """
         exprs = parse_positional_args_to_list(*cols)
         if not exprs:
             raise ValueError("The input of select() cannot be empty")
 
         names = []
+        table_func = None
+        join_plan = None
+
         for e in exprs:
             if isinstance(e, Column):
                 names.append(e._named())
             elif isinstance(e, str):
                 names.append(Column(e)._named())
+            elif isinstance(e, TableFunctionCall):
+                if table_func:
+                    raise ValueError(
+                        f"More than one table function cannot be called inside a select(). "
+                        f"Called '{table_func.name}' and '{e.name}'."
+                    )
+                else:
+                    table_func = e
+                    func_expr = _create_table_function_expression(func=table_func)
+                    join_plan = self._session._analyzer.resolve(
+                        TableFunctionJoin(self._plan, func_expr)
+                    )
```

Comment:
Out of curiosity: can we join table function with a `join_plan`? And if that is the case, do we still have "one table function" limitation?

Reply:
We can do a join in select when where the dataframe results from another join. This is possible because we know the hierarchy in which the joins are applied. An example I tried to test it is below:

```
>>> df.show()
---------------------------------------------------------
|"NAME"   |"LANGUAGES"     |"PROPERTIES"                |
---------------------------------------------------------
|Danilar  |[               |{                           |
|         |  "Alethi",     |  "alias": "Blackthorn",    |
|         |  "Azir",       |  "eye": "green",           |
|         |  "Old Tongue"  |  "hair": "black"           |
|         |]               |}                           |
|Kaladin  |[               |{                           |
|         |  "Alethi",     |  "alias": "Stormblessed",  |
|         |  "Java",       |  "eye": "brown/blue",      |
|         |  null          |  "hair": "brown"           |
|         |]               |}                           |
|Shallan  |[               |{                           |
|         |  "Alethi",     |  "alias": "Veil",          |
|         |  "Jah Kevedi"  |  "eye": "pale blue",       |
|         |]               |  "hair": "red"             |
|         |                |}                           |
|Stick    |NULL            |NULL                        |
|Vasher   |[               |{}                          |
|         |  "Alethi",     |                            |
|         |  "Nalthian"    |                            |
|         |]               |                            |
---------------------------------------------------------

>>> class MapUDTF:
...    def process(self, mp: dict) -> Tuple[str, str]:
...        if mp is None:
...            return None
...        for key in mp:
...            yield (key, mp[key])
>>> map_udtf = udtf(MapUDTF, output_schema=StructType([StructField("key", StringType()), StructField("value", StringType())]), input_types=[MapType()], name="a_very_creative_name", replace=True)
>>> class TransformerUDTF:
...     def process(self, name: str) -> str:
...         yield (name.upper(),)
>>> transformer_udtf = udtf(TransformerUDTF, output_schema=StructType([StructField("capital_name", StringType())]), input_types=[StringType()], name="captalist", replace=True)
>>> df2 = df.join_table_function(transformer_udtf(df.name))
>>> df2.select(df2.name, map_udtf(df2.properties), df2.capital_name).show()
---------------------------------------------------
|"NAME"   |"KEY"  |"VALUE"       |"CAPITAL_NAME"  |
---------------------------------------------------
|Kaladin  |alias  |Stormblessed  |KALADIN         |
|Kaladin  |eye    |brown/blue    |KALADIN         |
|Kaladin  |hair   |brown         |KALADIN         |
|Danilar  |alias  |Blackthorn    |DANILAR         |
|Danilar  |eye    |green         |DANILAR         |
|Danilar  |hair   |black         |DANILAR         |
|Shallan  |alias  |Veil          |SHALLAN         |
|Shallan  |eye    |pale blue     |SHALLAN         |
|Shallan  |hair   |red           |SHALLAN         |
---------------------------------------------------

```

Reply:
I think we should document it in the docstring (you can have `Note` section).